crates/evm/src/fuzz/mod.rs

```diff
@@ -216,6 +213,61 @@ impl<'a> FuzzedExecutor<'a> {
 
         result
     }
+
+    pub fn single_fuzz(
+        &self,
+        state: &EvmFuzzState,
+        address: Address,
+        should_fail: bool,
+        calldata: ethers::types::Bytes,
+    ) -> Result<FuzzOutcome, TestCaseError> {
+        let call = self
+            .executor
+            .call_raw(self.sender, address, calldata.0.clone(), 0.into())
+            .map_err(|_| TestCaseError::fail(FuzzError::FailedContractCall))?;
+        let state_changeset = call
+            .state_changeset
+            .as_ref()
+            .ok_or_else(|| TestCaseError::fail(FuzzError::EmptyChangeset))?;
+
+        // Build fuzzer state
+        collect_state_from_call(
+            &call.logs,
+            state_changeset,
+            state.clone(),
```

Comment:
Wondering, do we need to clone here? unsure if we need to or we can just borrow

Reply:
I think that we need to revert to the initial state for every step ? It's this way on master as well https://github.com/foundry-rs/foundry/blob/master/crates/evm/src/fuzz/mod.rs#L117